File: ParcelableDemo/app/src/main/java/com/example/parcelabledemo/detail/DetailActivity.java
package com.example.parcelabledemo.detail;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

import com.example.parcelabledemo.R;
import com.example.parcelabledemo.model.Details;

import java.util.List;

public class DetailActivity extends AppCompatActivity {

    TextView text_name, text_tech, text_location;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        text_name = findViewById(R.id.name);
        text_tech = findViewById(R.id.tech);
        text_location = findViewById(R.id.location);

        Intent intent = getIntent();
        List<Details> user = intent.getParcelableArrayListExtra("Information");

        text_name.setText("Name: " + user.get(0).getName());
        text_tech.setText("Technology: " + user.get(0).getTech());
        text_location.setText("Location: " + user.get(0).getLocation());

    }
}

File: NasaGallery/app/src/main/java/com/example/nasagallery/detail/DetailActivity.kt
package com.example.nasagallery.detail

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.example.nasagallery.R

class DetailActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        val title = intent.extras?.getString("title")
        val location = intent.extras?.getString("location")
        val photographer = intent.extras?.getString("photographer")
        val description = intent.extras?.getString("description")

        val titleView: TextView = findViewById(R.id.title)
        val locationView: TextView = findViewById(R.id.location)
        val photographerView: TextView = findViewById(R.id.photographer)
        val descriptionView: TextView = findViewById(R.id.description)
        val imageView: ImageView = findViewById(R.id.image)

        titleView.text = title
        locationView.text = location
        photographerView.text = photographer
        descriptionView.text = description

        Glide.with(this)
            .load(intent.extras?.getString("imageLink"))
            .into(imageView)
    }
}